sqlglot/dialects/exasol.py

```diff
@@ -394,6 +394,8 @@ def datatype_sql(self, expression: exp.DataType) -> str:
             # https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/to_timestamp.htm
             exp.Timestamp: rename_func("TO_TIMESTAMP"),
             exp.Quarter: lambda self, e: f"CEIL(MONTH(TO_DATE({self.sql(e, 'this')}))/3)",
+            # https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/rank.htm
+            exp.Rank: lambda *_: "RANK()",
```

Comment:
Let's remove this, add `rank_sql` under `Generator` and call `self.unsupported` if the rank has an argument.